### PR TITLE
feat: add iframe support

### LINKED ISSUE #68: Add `iframe` support
## Overview

Officially, [iframe](https://microlink.io/docs/api/parameters/iframe) has landed into Microlink API 🎉.

Now is the turn to leverage the functionality into SDK.

In order to enable it, just need to pass `iframe: true` as query parameter

```js
 const { status, data, response } = await mql(
    'https://www.youtube.com/watch?v=9P6rdqiybaw', {
      iframe: true
  })
```

Additionally, iframe options can be passed as an object:

```js
 const { status, data, response } = await mql(
    'https://www.youtube.com/watch?v=9P6rdqiybaw', {
      iframe: { width: 480, height: 270 }
  })
```

This can be used by some provider in order to customize the output

After that,  the payload returns a new data field called `iframe` with the embeddable code to be inserted as markup, e.g.:

```json
{
  "iframe": "<iframe width=\"480\" height=\"270\" src=\"https://www.youtube.com/embed/9P6rdqiybaw?feature=oembed\" frameborder=\"0\" allow=\"accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen></iframe>"
}
```

## Requeriments

Implement `iframe` in a way that, when is present, the card output will be native iframe provided by the service instead of the common Microlink card visualization. 

```jsx
<Microlink 
  url='https://www.youtube.com/watch?v=9P6rdqiybaw' 
  media='iframe' 
/>
````

### iframe is not widely supported by internet

Most of the URLs don't return an iframe and it isn't possible to determinate _before_ performing the API call. 

That means `iframe` support needs to support the common card visualization as a fallback.

I suggest keep using `media` prop for that:

```jsx
// fallback into image card visualization
// if iframe doesn't present.
<Microlink 
  url='https://example.com' 
  media={['iframe', 'image']}
/>
```

Although a new prop can be added:

```jsx
// fallback into image card visualization
// if iframe doesn't present.
<Microlink 
  url='https://example.com' 
  preferMode={['iframe', 'card']}
/>
```

In fact, need to find a way to be possible pass iframe settings

```jsx
// fallback into image card visualization
// if iframe doesn't present.
<Microlink 
  url='https://example.com' 
  preferMode={['iframe', 'card']}
  iframe={{ width 480, height: 270 }}
/>
```

If we need to choose, I prefer to don't extend the current API surface and find a way to accommodate the functionality with the things we have.

### iframe is not actually an iframe

You can't believe `iframe` is going to be the root HTML element node all the time.

e.g., consider the following API call: https://api.microlink.io/?url=https://view.genial.ly/5dc53cfa759d2a0f4c7db5f4&iframe

The root node here is a wrapper for control the iframe responsive ¯\_(ツ)_/¯

### iframe skeleton loader time

As the same way we are showing a loading skeleton until the data is fetched from API properly:

demo

We need to do the same until the iframe markup is mounted and ready for displaying the media.

This can be done adding an event listener to `iframe` HTML element: https://github.com/microlinkhq/www/blob/master/src/components/elements/IframeInline/IframeInline.js#L16

We can reuse the current card skeleton loader we have for the card since actually we don't know if we can mount an iframe until the API call is performed.

### PR DESCRIPTION
**Pull request by Kikobeats**
Closes #68

### CODE REVIEW

File: packages/react/src/index.js [hunk 1]
```diff
@@ -206,6 +233,7 @@ Microlink.propTypes = {
   contrast: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
   controls: PropTypes.bool,
   direction: PropTypes.string,
+  iframe: PropTypes.string,
```
Comment (breadadams) [inline packages/react/src/index.js]: Is this prop still used @Kikobeats?

File: packages/react/src/utils/index.js [hunk 2]
```diff
@@ -43,8 +52,8 @@ export const getApiUrl = ({
   data,
   force,
   headers,
+  iframe,
   media,
-  prerender = 'auto',
```
Comment (breadadams) [inline packages/react/src/utils/index.js]: Just out of interest, how come `prerender` has been removed?

Comment (Kikobeats) [reply packages/react/src/utils/index.js]: I removed it because 'prerender=auto' is the default API value

File: packages/react/src/index.js (hunk 1 above)
Comment (Kikobeats) [reply packages/react/src/index.js]: nope, good catch!

File: packages/react/src/utils/index.js (hunk 2 above)
Comment (breadadams) [reply packages/react/src/utils/index.js]: Ah right, but then you should just remove the `= 'auto'` part right? We still want to pass `prerender` to `getApiUrl`.

Comment (Kikobeats) [reply packages/react/src/utils/index.js]: oh, totally true!

**Pull request merged**